packages/ts/react-crud/src/property-info.ts

```diff
@@ -59,34 +60,47 @@ export function _generateHeader(path: string): string {
     .replace(/^./u, (match) => match.toUpperCase());
 }
 
-export const getPropertyIds = (model: DetachedModelConstructor<AbstractModel>): string[] =>
-  Object.keys(Object.getOwnPropertyDescriptors(model.prototype)).filter((p) => p !== 'constructor');
+export const getPropertyIds = (model: DetachedModelConstructor<AbstractModel>): string[] => {
+  const propertyIds: string[] = [];
+
+  for (let proto = model; proto !== ObjectModel; proto = Object.getPrototypeOf(proto)) {
+    // the choice of putting parent properties at the beginning is deliberate, `push` could be used instead
+    propertyIds.unshift(...Object.keys(Object.getOwnPropertyDescriptors(proto.prototype)).filter((p) => p !== 'new'));
+  }
+
+  return propertyIds;
+};
 
 export const getProperties = (model: DetachedModelConstructor<AbstractModel>): PropertyInfo[] => {
   const propertyIds = getPropertyIds(model);
   // eslint-disable-next-line @typescript-eslint/no-unsafe-call
   const modelInstance: any = createDetachedModel(model);
-  return propertyIds.flatMap((name) => {
-    // eslint-disable-next-line
-    const propertyModel = modelInstance[name] as AbstractModel;
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    const meta = propertyModel[_meta];
-    const humanReadableName = _generateHeader(name);
-    const type = determinePropertyType(propertyModel);
-
-    if (hasAnnotation(meta, 'jakarta.persistence.OneToOne')) {
-      // Expand sub properties
-      const subProps = getProperties(propertyModel.constructor as any);
-      return subProps.map((prop) => ({ ...prop, name: `${name}.${prop.name}` }));
-    }
-
-    return {
-      name,
-      humanReadableName,
-      type,
-      meta,
-    };
-  });
+  return (
+    propertyIds
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      .map((name) => [name, modelInstance[name]])
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      .filter(([, m]) => m?.[_meta])
```

Comment:
Is the filter here necessary? Metadata is in general optional and there are specific conditions that determine whether it is generated at all.

Reply:
I'm filtering out those who don't have metadata at all

Reply:
The original code (before this PR) assumes that `meta` is there. It maybe is, but TypeScript complains if I remove this check.

Reply:
I just tried to remove it, I can not observe any errors.